runelite-client/src/main/java/net/runelite/client/plugins/loginscreen/LoginScreenPlugin.java

```diff
@@ -277,6 +279,22 @@ private void overrideLoginScreen()
 				}
 			}
 		}
+		else if (config.loginScreen() == LoginScreenOverride.RANDOM)
+		{
+
+			ArrayList<String> loginScreenOverrides = new ArrayList<>();
```

Comment:
You can just use the values array.

Reply:
OFF, CUSTOM and RANDOM don't have a associated file name so I build a new list leaving them out.

Reply:
The enum is in a specific order.

Reply:
I don't see how the order affects things.

Reply:
The order isn't important, but there is no need to create a new list for this when you can `Arrays.stream()` the values and choose a random from that, via `Stream.skip(randomInt)` for instance.

Reply:
```java
/**
* Returns an array containing the constants of this enum   
* type, in the order they're declared.  This method may be  
* used to iterate over the constants as follows:  
*  
*    for(E c : E.values())  
*        System.out.println(c);  
*  
* @return an array containing the constants of this enum   
* type, in the order they're declared  
*/  
public static E[] values();
```

Then you can do

```
int idx = ThreadLocalRandom.current().nextInt(1, enum.values().length - 1);
//or
int idx = new Random().nextInt(enum.values().length - 1) + 1;
String fileName = enum.values()[idx];
```

Reply:
> ```
> int idx = ThreadLocalRandom.current().nextInt(1, enum.values().length - 1);
> //or
> int idx = new Random().nextInt(enum.values().length - 2) + 1;
> String fileName = enum.values()[idx];
> ```

Personally I don't like the idea of using magic numbers to filter out the the first and last 2 elements of the enum.

Reply:
Those aren't magic numbers.